src/odemis/driver/actuator.py

```diff
@@ -1907,7 +1896,48 @@ def _doMoveAbs(self, pos):
         else:
             move = -mod2
 
-        self._move_sum += abs(move)
+        # Check that the move passes by the reference switch by detecting whether
+        # the current and final position are not in the same multiple of cycle.
+        # "Linear" view of the axis (c = cycle)
+        #    -2c       -1c        0         c         2c        3c
+        # ----|---------|---------|---------|---------|---------|---
+        #         -2            -1      0        1        2
+        final_pos = cur_pos + move
+        pass_ref = (cur_pos // self._cycle) != (final_pos // self._cycle)
+        
+        if pass_ref or self._move_num_total >= 5:
+            # Move to pos close to ref switch
+            # Optimize by moving through the closest way
+            cur_pos = self._child.position.value[self._caxis]
```

Comment:
As discussed, please refactor the computation of the shortest move to a separate function.